libs/application-generic/src/usecases/workflow/delete-workflow/delete-workflow.usecase.ts

```diff
@@ -0,0 +1,136 @@
+import { Injectable } from '@nestjs/common';
+
+import {
+  ControlValuesRepository,
+  MessageTemplateRepository,
+  NotificationTemplateEntity,
+  NotificationTemplateRepository,
+  PreferencesRepository,
+} from '@novu/dal';
+import { PreferencesTypeEnum, WorkflowOriginEnum } from '@novu/shared';
+
+import { DeleteWorkflowCommand } from './delete-workflow.command';
+import { InvalidateCacheService } from '../../../services/cache/invalidate-cache.service';
+import { GetWorkflowByIdsUseCase } from '../get-workflow-by-ids/get-workflow-by-ids.usecase';
+import { GetWorkflowByIdsCommand } from '../get-workflow-by-ids/get-workflow-by-ids.command';
+import {
+  buildNotificationTemplateIdentifierKey,
+  buildNotificationTemplateKey,
+} from '../../../services/cache/key-builders';
+
+@Injectable()
+export class DeleteWorkflowUseCase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private messageTemplateRepository: MessageTemplateRepository,
+    private getWorkflowByIdsUseCase: GetWorkflowByIdsUseCase,
+    private preferencesRepository: PreferencesRepository,
+    private invalidateCache: InvalidateCacheService,
+    private controlValuesRepository: ControlValuesRepository,
+  ) {}
+
+  async execute(command: DeleteWorkflowCommand): Promise<void> {
+    const workflowEntity = await this.getWorkflowByIdsUseCase.execute(
+      GetWorkflowByIdsCommand.create({
+        ...command,
+        identifierOrInternalId: command.identifierOrInternalId,
+      }),
+    );
+
+    await this.deleteRelatedEntities(command, workflowEntity);
+  }
+
+  private async deleteRelatedEntities(
```

Comment:
@djabarovgeorge we need a transaction here in the same fashion we did it [here](https://github.com/novuhq/novu/blob/5a0efd7f02cb4c579c65439751daefa029cc551e/apps/api/src/app/subscribers/usecases/remove-subscriber/remove-subscriber.usecase.ts#L58). to ensure that all of these are deleted atomically.

Reply:
done